### ISSUE TITLE
GOG sourced z-code games do not work.

### ISSUE THREAD

**Issue by NULUSIOS** [garglk/garglk#241]
Those are Z-Code, and have .dat extension (although this doesn't matter, I tried using .z6 or something).
SCUMMVM very old gargoyle implementation can run it, so I suspect it is a trivial issue (some header to ignore or something?).

**Comment by salty-horse** [comment]
Which games are you trying in particular?

The Zork Anthology works for me after renaming the files to have a ".z#" extension (the number doesn't seem to matter).

ScummVM doesn't have official Z-Code support (there was an outside project working on adding support, but it was never merged).

**Comment by cspiegel** [comment]
If you've got access to a UNIX-like system with hexdump, can you include the output of the following command for one of the games

`hexdump -n 64 -C blah.dat`

And include which game this is?

Or if you have any other way (e.g. with a hex editor) of sending a hex dump of the first 64 bytes, that would be helpful in determining if these look like proper Z-machine games or not.

**Comment by NULUSIOS** [comment]
Here is an example complete file (provided for documentation and testing):
https://www.dropbox.com/s/2aw3cvxoosz08rl/beyondzork-gog.zip?dl=0

**Comment by cspiegel** [comment]
It works fine for me (after renaming to .z5).

What version of Gargoyle are you trying this on, and what operating system?  When you say "SCUMMVM very old gargoyle", what do you mean?  Gargoyle and ScummVM are completely unrelated.  If you're trying to use ScummVM to run this, it just won't work.

**Comment by NULUSIOS** [comment]
Windows.
There were old test builds of SCUMMVM with Gargoyle engine embedded.
I can send you that too.

**Comment by cspiegel** [comment]
OK, but are you using Gargoyle by itself, or this ScummVM+Gargoyle build (which looks like it may be [ScummVM-Misc](http://scummvm-misc.sourceforge.net/))?  If you're using ScummVM at all, bug reports for it do not belong here.  ScummVM-Misc seems to have a bug tracker [here](https://sourceforge.net/p/scummvm-misc/bugs/), but it seems that the project isn't really maintained, at least not the Gargoyle integration.

In any case, if you're not using Gargoyle directly, this issue tracker is the wrong place for bug reports.

**Comment by NULUSIOS** [comment]
Again: I am talking about Gargoyle, else I wouldn't post here.
I say Gargoyle doesn't run these versions, while ScummVM/Gargoyle build, does.

**Comment by cspiegel** [comment]
Again: I asked what version of Gargoyle you are trying this on.

**Comment by NULUSIOS** [comment]
Right, sorry.
A compile I found dated Oct 23rd 2016.
This: https://www.dropbox.com/s/cpgl9btx97a1cd5/gargoyle_23-oct-2016.zip?dl=0

Also here is the (modernized by a dev), scummvm + gargoyle that can run GOG/z-code games:
https://www.dropbox.com/s/5b9z5y1tsr89s1x/Gargoyle-aviloria_mod.7z?dl=0

Question is, I provided a problematic game. You have a fresh compile where it works?

**Comment by cspiegel** [comment]
OK, thank you.

I don't have access to a Windows machine at the moment, but I successfully ran the provided game under the provided Gargoyle build using Wine on Linux.  It ran fine with gargoyle (which calls Bocfel), and also ran fine with the other Z-machine interpreters, Frotz and Nitfol.

I have an old Subversion r612 build that it works with, and I built the latest Git version, and it runs there too.

So I'm not sure why you're having trouble.  What happens (including specific error messages, if any) when you try to run it?
